src/main/java/fr/insee/keycloak/provider/FranceConnectIdentityProvider.java

```diff
@@ -103,7 +104,19 @@ protected boolean verify(JWSInput jws) {
     @Override
     public BrokeredIdentityContext getFederatedIdentity(String response) {
         try {
-            return super.getFederatedIdentity(response);
+            BrokeredIdentityContext federatedIdentity = super.getFederatedIdentity(response);
+            // Add verification for eidas level
+            JsonWebToken idToken = (JsonWebToken) federatedIdentity.getContextData().get("VALIDATED_ID_TOKEN");
+            FranceConnectIdentityProviderConfig.EidasLevel eidasLevelReturned = FranceConnectIdentityProviderConfig.EidasLevel.getOrDefault((String) idToken.getOtherClaims().get("acr"), null);
```

Comment:
We can create constants for "VALIDATED_ID_TOKEN" and "acr" to avoid "magic strings" ^^ and store `(String) idToken.getOtherClaims().get("acr")` in a variable to beautify this great code :)
Thanks for your changes.